### PR TITLE
Support event attribute isPublic

### PR DESCRIPTION
**Pull request by Lin1404**

### CODE REVIEW

File: src/main/java/com/timebook/timebook/models/events/Event.java [hunk 1]
```diff
@@ -33,14 +34,16 @@ public Event() {
...
+            int priority,
+            Boolean isVisible) {
         this.id = id;
         this.email = email;
         this.title = title;
         this.description = description;
         this.startDateTime = startDateTime;
         this.endDateTime = endDateTime;
         this.priority = priority;
+        this.isVisible = isVisible;
```
Comment (HaofengLiang) [inline src/main/java/com/timebook/timebook/models/events/Event.java]: I just think of a name "isPublic". Would it more accurately describe the attribute?

File: src/main/java/com/timebook/timebook/models/events/Event.java [hunk 2]
```diff
@@ -114,7 +125,8 @@ public boolean equals(Object o) {
                 && Objects.equals(this.description, event.description)
                 && Objects.equals(this.startDateTime, event.startDateTime)
                 && Objects.equals(this.endDateTime, event.endDateTime)
-                && Objects.equals(this.priority, priority);
+                && Objects.equals(this.priority, priority)
```
Comment (HaofengLiang) [inline src/main/java/com/timebook/timebook/models/events/Event.java]: Do these 2 need to be in the equals method?

File: src/main/java/com/timebook/timebook/service/EventService.java [hunk 3]
```diff
@@ -27,14 +29,27 @@ public EventService(EventRepository eventRepository, UserService userService) {
...
+            throw new ValidationException("Event validation failed.");
+        }
         return eventRepository.save(requestEvent);
     }
 
     public void delete(long id) {
         eventRepository.deleteById(id);
     }
 
+    private boolean isEventValid(Event event) {
```
Comment (HaofengLiang) [inline src/main/java/com/timebook/timebook/service/EventService.java:39]: nit: I would prefer this move method close to where it's used.

**Review by weixiang237 — APPROVED** [review]
lgtm

**Pull request merged**